arviz/plots/rankplot.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+
+from ..data import convert_to_dataset
+from .plot_utils import (
+    _scale_fig_size,
+    xarray_var_iter,
+    default_grid,
+    _create_axes_grid,
+    make_label,
+)
+from ..utils import _var_names
+
+
+def plot_rank(data, var_names=None, coords=None, bins="auto", figsize=None, axes=None):
+
+    posterior_data = convert_to_dataset(data, group="posterior")
+    if coords is not None:
+        posterior_data = posterior_data.sel(**coords)
+    var_names = _var_names(var_names, posterior_data)
+    plotters = list(xarray_var_iter(posterior_data, var_names=var_names, combined=True))
+
+    if axes is None:
+        rows, cols = default_grid(len(plotters))
+
+        figsize, ax_labelsize, titlesize, _, _, _ = _scale_fig_size(
+            figsize, None, rows=rows, cols=cols
+        )
+        _, axes = _create_axes_grid(len(plotters), rows, cols, figsize=figsize, squeeze=False)
+
+    for ax, (var_name, selection, var_data) in zip(axes.ravel(), plotters):
+        ranks = var_data.argsort(axis=None).reshape(*var_data.shape)
+        all_counts = []
+        for row in ranks:
+            # If `bins` was auto, it gets overwritten here, and carried forward
+            # using the `range` argument makes sure the bins cover all the data
+            counts, bins = np.histogram(row, bins=bins, range=(0, ranks.size))
+            all_counts.append(counts)
+        all_counts = np.array(all_counts)
+        gap = all_counts.max() * 1.05
+        width = (bins[1] - bins[0]) * 0.95
+
+        y_ticks = []
+        for idx, counts in enumerate(all_counts):
+            y_ticks.append(idx * gap)
+            # Line where data is uniform
+            ax.axhline(
+                y=y_ticks[-1] + counts.sum() / bins.shape[0], linestyle="--", color="C1", zorder=-10
```

Comment:
Should this be `np.mean(counts)`, because counts.size = bins.size - 1

Reply:
Also, I think this line look better on top.

This should probably be optional ?

Reply:
optional is a good idea, as is `np.mean`. I like it better in back, but you've made a lot of good points here, so i'm trusting you!